### ISSUE TITLE
bareli - wrong  calling initialize  function to "address(0)"

### ISSUE THREAD

**Issue by sherlock-admin3** [sherlock-audit/2024-02-optimism-2024-judging#165]
bareli

medium

# wrong  calling initialize  function to "address(0)"

## Summary
Here we are calling initialize function  in constructor with address(0). This will set the initialize function according to value assign in constructor. we are using" initializer" from "openzepplin" so we can call   initialize function only once.

## Vulnerability Detail
    constructor(uint256 _delay) {
        DELAY_SECONDS = _delay;
   @>>     initialize({ _owner: address(0), _config: SuperchainConfig(address(0)) });
    }

    function initialize(address _owner, SuperchainConfig _config) public initializer {
        __Ownable_init();
        _transferOwnership(_owner);
        config = _config;
    }

   

   >>>   _transferOwnership(_owner); will behave like a renouncing ownership will leave the contract without an owner,
     thereby disabling any functionality that is only available to the owner as shown below.we cannot use recover and hold function.
  
     
   //                from openzeppelin
    function _transferOwnership(address newOwner) internal virtual {
        OwnableStorage storage $ = _getOwnableStorage();
        address oldOwner = $._owner;
        $._owner = newOwner;
        emit OwnershipTransferred(oldOwner, newOwner);
    }

   function __Ownable_init(address initialOwner) internal onlyInitializing {
        __Ownable_init_unchained(initialOwner);
    }
 function __Ownable_init_unchained(address initialOwner) internal onlyInitializing {
        if (initialOwner == address(0)) {
            revert OwnableInvalidOwner(address(0));
        }
        _transferOwnership(initialOwner);
    }

## Impact
we cannot use hold and recover function.
## Code Snippet
https://github.com/sherlock-audit/2024-02-optimism-2024/blob/main/optimism/packages/contracts-bedrock/src/dispute/weth/DelayedWETH.sol#L39
## Tool used

Manual Review

## Recommendation
  constructor(uint256 _delay,address owner) {
        DELAY_SECONDS = _delay;
   @>>     initialize({ _owner: owner), _config: SuperchainConfig(owner) });
    }

Duplicate of #38
